metomi/rosie/suite_id.py

```diff
@@ -365,9 +364,31 @@ def _from_id_text(self, id_text):
 
     def _from_location(self, location):
         """Return the ID of a location (origin URL or local copy path)."""
-        # Is location a "~/cylc-run/$SUITE/" directory?
-        # Use a hacky way to read the "log/rose-suite-run.version" file
         suite_engine_proc = SuiteEngineProcessor.get_processor()
+        suite_dir_rel_root = getattr(
+            suite_engine_proc, "SUITE_DIR_REL_ROOT", None)
+
+        # Cylc8 run directory
+        # TODO: extract version control information
+        loc = Path(location)
+        sdrr = Path('~', suite_dir_rel_root).expanduser().resolve()
+        try:
+            if loc.relative_to(sdrr):
+                if (loc / 'rose-suite.info').exists():
+                    # This is an installed workflow with a rose-suite.info file
+                    # (most likely a Cylc8 run directory)
+
+                    # TODO: extract version control information written by
+                    # Cylc install, see:
+                    # https://github.com/metomi/rose/issues/2432
+                    # https://github.com/cylc/cylc-flow/issues/3849
+                    raise SuiteIdLocationError(location)
+        except ValueError:
+            # Not an installed Cylc8 workflow run directory
+            pass
```

Comment:
I think `if Path('')` evaluates as `if True`, so better to remove the `if` completely to avoid confusion
```suggestion
        try:
            loc.relative_to(sdrr)
        except ValueError:
            # Not an installed Cylc8 workflow run directory
            pass
        else:
            if (loc / 'rose-suite.info').is_file():
                # This is an installed workflow with a rose-suite.info file
                # (most likely a Cylc8 run directory)

                # TODO: extract version control information written by
                # Cylc install, see:
                # https://github.com/metomi/rose/issues/2432
                # https://github.com/cylc/cylc-flow/issues/3849
                raise SuiteIdLocationError(location)
```

Reply:
Good spot.

Ah, yes, I was using it like `is_relative_to` (which we can't use because it was introduced in Python3.9).